app/src/main/java/chat/rocket/android/api/rest/CookieInterceptor.java

```diff
@@ -0,0 +1,27 @@
+package chat.rocket.android.api.rest;
+
+import java.io.IOException;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class CookieInterceptor implements Interceptor {
+
+  private final CookieProvider cookieProvider;
+
+  public CookieInterceptor(CookieProvider cookieProvider) {
+    this.cookieProvider = cookieProvider;
+  }
+
+  @Override
+  public Response intercept(Chain chain) throws IOException {
+    if (chain.request().url().host().equals(cookieProvider.getHostname())) {
```

Comment:
This logic probably doesn't work when multiple server configuration exists (in future).
Some kind of iteration for ` [host] => [cookie]` would be required here.